projects/packages/videopress/src/class-data.php

```diff
@@ -129,9 +135,43 @@ public static function has_connected_owner() {
 	 * @return array
 	 */
 	public static function get_initial_state() {
+		$videopress_data   = self::get_video_data();
+		$local_videos_data = self::get_video_data( false );
 
-		$video_data = self::get_video_data();
+		// Tweak local videos data.
+		$local_videos = array_map(
+			function ( $video ) {
+				$id                 = $video['id'];
+				$media_details      = $video['media_details'];
+				$jetpack_videopress = $video['jetpack_videopress'];
 
+				$upload_date = $video['date'];
+				$url         = $video['source_url'];
+
+				$title       = $jetpack_videopress['title'];
+				$description = $jetpack_videopress['description'];
+				$caption     = $jetpack_videopress['caption'];
+
+				$width    = $media_details['width'];
+				$height   = $media_details['height'];
+				$duration = $media_details['length'];
+
+				return array(
+					'id'          => $id,
+					'title'       => $title,
+					'description' => $description,
+					'caption'     => $caption,
+					'width'       => $width,
+					'height'      => $height,
+					'url'         => $url,
+					'uploadDate'  => $upload_date,
+					'duration'    => $duration,
+				);
+			},
+			$local_videos_data['videos']
+		);
+
+		// Tweak local videos data.
```

Comment:
Actually, here we are dealing with VideoPress data:

```suggestion
		// Tweak VideoPress videos data.
```